README.rst
Python Client for Cloud Data Fusion API
=======================================

|stable| |pypi| |versions|

`Cloud Data Fusion API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-data-fusion.svg
   :target: https://pypi.org/project/google-cloud-data-fusion/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-data-fusion.svg
   :target: https://pypi.org/project/google-cloud-data-fusion/
.. _Cloud Data Fusion API: https://cloud.google.com/data-fusion
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/datafusion/latest
.. _Product Documentation:  https://cloud.google.com/data-fusion

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Cloud Data Fusion API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Cloud Data Fusion API.:  https://cloud.google.com/data-fusion
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-data-fusion


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-data-fusion

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Cloud Data Fusion API
   to see other available methods on the client.
-  Read the `Cloud Data Fusion API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Cloud Data Fusion API Product documentation:  https://cloud.google.com/data-fusion
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

## [1.3.0](https://github.com/googleapis/python-data-fusion/compare/v1.2.1...v1.3.0) (2022-01-18)


### Features

* update definitions for cloud/datafusion/v1 and cloud/datafusion/v1beta1  ([#58](https://github.com/googleapis/python-data-fusion/issues/58)) ([6b38819](https://github.com/googleapis/python-data-fusion/commit/6b38819f26fb72dc67ac2a4dda1c543d91b7f835))

## [1.3.0](https://github.com/googleapis/python-data-fusion/compare/v1.2.1...v1.3.0) (2022-01-14)


### Features

* update definitions for cloud/datafusion/v1 and cloud/datafusion/v1beta1  ([#58](https://github.com/googleapis/python-data-fusion/issues/58)) ([6b38819](https://github.com/googleapis/python-data-fusion/commit/6b38819f26fb72dc67ac2a4dda1c543d91b7f835))

### [1.2.1](https://www.github.com/googleapis/python-data-fusion/compare/v1.2.0...v1.2.1) (2021-11-01)


### Bug Fixes

* **deps:** drop packaging dependency ([b4d2c35](https://www.github.com/googleapis/python-data-fusion/commit/b4d2c355353bb2621f2c077a15f2505068718fc1))
* **deps:** require google-api-core >= 1.28.0 ([b4d2c35](https://www.github.com/googleapis/python-data-fusion/commit/b4d2c355353bb2621f2c077a15f2505068718fc1))


### Documentation

* list oneofs in docstring ([b4d2c35](https://www.github.com/googleapis/python-data-fusion/commit/b4d2c355353bb2621f2c077a15f2505068718fc1))

## [1.2.0](https://www.github.com/googleapis/python-data-fusion/compare/v1.1.0...v1.2.0) (2021-10-15)


### Features

* add support for python 3.10 ([#39](https://www.github.com/googleapis/python-data-fusion/issues/39)) ([09aefb0](https://www.github.com/googleapis/python-data-fusion/commit/09aefb0e13a748a7052f3bdf87fe4341c06fc28a))

## [1.1.0](https://www.github.com/googleapis/python-data-fusion/compare/v1.0.2...v1.1.0) (2021-10-08)


### Features

* add context manager support in client ([#36](https://www.github.com/googleapis/python-data-fusion/issues/36)) ([3c238d5](https://www.github.com/googleapis/python-data-fusion/commit/3c238d5d26f219dd107f09dcec3fc09977d64760))

### [1.0.2](https://www.github.com/googleapis/python-data-fusion/compare/v1.0.1...v1.0.2) (2021-10-04)


### Bug Fixes

* improper types in pagers generation ([b278e83](https://www.github.com/googleapis/python-data-fusion/commit/b278e83f4a087ac21fd02eefc5f79e8c02abcfb5))

### [1.0.1](https://www.github.com/googleapis/python-data-fusion/compare/v1.0.0...v1.0.1) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([73e18c4](https://www.github.com/googleapis/python-data-fusion/commit/73e18c4504ee8f1f86d9122b4fb8db223ee24cff))

## [1.0.0](https://www.github.com/googleapis/python-data-fusion/compare/v0.1.2...v1.0.0) (2021-08-24)


### Features

* bump release level to production/stable ([#14](https://www.github.com/googleapis/python-data-fusion/issues/14)) ([26f5506](https://www.github.com/googleapis/python-data-fusion/commit/26f5506b24453c6764f41637c18c6ceb10a9ba3c))


### Documentation

* migrate to main branch ([#20](https://www.github.com/googleapis/python-data-fusion/issues/20)) ([7edab48](https://www.github.com/googleapis/python-data-fusion/commit/7edab48370aeb6194f864bc2d402b8ffa7761a51))

### [0.1.2](https://www.github.com/googleapis/python-data-fusion/compare/v0.1.1...v0.1.2) (2021-07-29)


### Bug Fixes

* enable self signed jwt for grpc ([#10](https://www.github.com/googleapis/python-data-fusion/issues/10)) ([0ec8226](https://www.github.com/googleapis/python-data-fusion/commit/0ec82261f2c4fab58a2a52ec9d3c49d043598f2c))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#6](https://www.github.com/googleapis/python-data-fusion/issues/6)) ([f97a1a8](https://www.github.com/googleapis/python-data-fusion/commit/f97a1a8e7d7ffdac8cbf7c7364e5b6797f818e4d))


### Miscellaneous Chores

* release as 0.1.2 ([#11](https://www.github.com/googleapis/python-data-fusion/issues/11)) ([6b418a0](https://www.github.com/googleapis/python-data-fusion/commit/6b418a0d333f81771a597e0a554d2bf05b31d962))

### [0.1.1](https://www.github.com/googleapis/python-data-fusion/compare/v0.1.0...v0.1.1) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#5](https://www.github.com/googleapis/python-data-fusion/issues/5)) ([bcc8d82](https://www.github.com/googleapis/python-data-fusion/commit/bcc8d8292b2d474ee504483707d7856af9ecf7e0))

## 0.1.0 (2021-07-06)


### Features

* generate v1 ([8dff153](https://www.github.com/googleapis/python-data-fusion/commit/8dff15325e970ee1fbf09952026c47235f0ed8e7))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9', '3.10'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/data_fusion_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.9')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.9')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
